SimPPS/RPDigiProducer/plugins/RPDigiProducer.cc

```diff
@@ -0,0 +1,279 @@
+// user include files
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "SimDataFormats/TrackingHit/interface/PSimHitContainer.h"
+#include "SimDataFormats/TrackingHit/interface/PSimHit.h"
+#include "SimDataFormats/CrossingFrame/interface/CrossingFrame.h"
+#include "SimDataFormats/CrossingFrame/interface/MixCollection.h"
+//needed for the geometry:
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+#include "DataFormats/CTPPSDigi/interface/TotemRPDigi.h"
+#include "DataFormats/CTPPSDigi/interface/TotemRPDigi.h"
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/CTPPSDetId/interface/TotemRPDetId.h"
+
+//Random Number
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Utilities/interface/RandomNumberGenerator.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "CLHEP/Random/RandomEngine.h"
+#include "CondFormats/CTPPSReadoutObjects/interface/TotemAnalysisMask.h"
+#include "CondFormats/DataRecord/interface/TotemReadoutRcd.h"
+#include "SimPPS/RPDigiProducer/interface/RPSimTypes.h"
+#include "SimPPS/RPDigiProducer/interface/RPDetDigitizer.h"
+
+#include "DataFormats/Common/interface/DetSet.h"
+#include "SimPPS/RPDigiProducer/interface/DeadChannelsManager.h"
+
+#include "boost/shared_ptr.hpp"
+
+// system include files
+#include <memory>
+#include <vector>
+#include <map>
+#include <string>
+#include <iostream>
+#include <cstdlib> // I need it for random numbers
+
+// user include files
+
+
+//
+// class decleration
+//
+
+namespace CLHEP {
+  class HepRandomEngine;
+}
+
+
+class RPDigiProducer : public edm::EDProducer {
```

Comment:
This should inherit from one of the threaded EDProducers, likely `stream` is the simplest one.